flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/InputChannelInfo.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.checkpoint.channel;
+
+import org.apache.flink.annotation.Internal;
+
+import java.io.Serializable;
+
+/**
+ * Identifies {@link org.apache.flink.runtime.io.network.partition.consumer.InputChannel} in a given subtask.
+ */
+@Internal
+public class InputChannelInfo implements Serializable {
```

Comment:
It's confusing that we now have `InputChannelID` and `InputChannelInfo`. I don't see a good solution though. Similarly, it's a bit of a pity that we cannot reuse this class for `ResultSubpartitionInfo` (virtually it's the same thing). Maybe @zhijiangW has an idea.

Reply:
Unfortunately, we can't use `ID`s as they are currently assigned at random. I'll put it into the javadoc.

To identify state, we logically need: `source task id`, `target task id`, `isUpstream`. Instead of `isUpstream`, I thought it's better to have an explicit entity for both `InputChannel` and `ResultSubpartition`.

Regarding names, it was actually @zhijiangW (good) idea:)

Reply:
It indeed brings confusing for having both `InputChannelID` and `InputChannelInfo` now.

`InputChannelID` is the unique id for identifying while requesting partitions from different tasks on upstream side.  
`InputChannelInfo` is a wrapper structure which would be easily extended to cover more infos (e.g. task index) future if necessary. And currently the strong requirement is to identify it for different `InputGate` instances.

The "inspiration" is form existing `TaskInfo` structure coincides with `executionId`. I also do not see a better solution yet.

Reply:
Let's stick with this structure then.

Reply:
Ideally, we could replace `InputChannelID` with Roman's `InputChannelInfo`, but that's probably too much of an effort. It would certainly make programs easier to debug.

Reply:
`InputChannelID` can not be replaced directly by InputChannelInfo` as I described the function above.

For every upstream task, it can receive the partition requests from the `InputChannel` of different task instances. And now it uses the global unique `InputChannelID` for distinguishing with different tasks.

But `InputChannelInfo` can only guarantee the uniqueness within one task scope,  and still has the same id among different tasks, unless we introduce `InputChannelID` into the `InputChannelInfo`.